### ISSUE TITLE
Can't Reproduce the Results

### ISSUE THREAD

**Issue by AGTSAAA** [acharkq/MolCA#3]
Hi, 

Thank you for sharing the code.

I have run the command  following your provided Readme.

`python stage1.py --root 'data/PubChem324k' --gtm --lm --devices '0,1' --mode train --filename stage1 --rerank_cand_num 128 --num_query_token 8 --tune_gnn --save_every_n_epochs` 

for the stage 1 training and save the model every 5 epochs. 

Then, I use the final epoch's model to run the evaluation code on the PCDes dataset using the following command 

`python stage1.py --root 'data/kv_data' --gtm --lm --devices '[0]'  --filename pcdes_evaluation --init_checkpoint "all_checkpoints/stage1/epoch=49.ckpt" --rerank_cand_num 128 --num_query_token 8 --match_batch_size 64 --mode eval`.
 
Then, I get the result shown in the image.

 I find  that I can't reproduce the results discussed in the paper.  For instance, the results test_fullset_t2g_rec20 is 79.55 test_fullset_g2t_rec20 79.22.  However,  you reported 87.6 and 90.5, respectively.

![28681699156635_ pic](https://github.com/acharkq/MolCA/assets/83741666/0e48d2e6-a623-48ba-bc05-80c7ba675bf3)

Can you help me to resolve this?

Thank you very much!

**Comment by acharkq** [comment]
Thanks for the interest! I am now looking into this problem.

Some details can help me to locate the problem:

* May I know if you can reproduce the results on the retrieval results of the PubChem324k dataset and the MoMu dataset?
* From your results, it seems the reranking module is not working at all. Can you provide the training loss and see if the ranking loss is decreasing?

**Comment by acharkq** [comment]
Oh, I find the problem.

It seems that the dataloading module was not loading the pretraining subset of PubChem324k correct, because of a naming issue. 

I have fixed this issue in the latest commit. Can you try again the stage 1 pretraining? Note that, this time the pretraining will take longer, because the previous code was wrongly using the smaller training set for stage 1 pretraining.

2d46afed78c935b5c7299d0af085e687f547131e

**Comment by acharkq** [comment]
I am closing this issue. Feel free to open it again if there is still problem on reproducing the results.
